src/back/hlsl/writer.rs

```diff
@@ -72,42 +92,267 @@ impl<W: Write> Writer<W> {
+
+        Ok(())
+    }
+
+    fn write_binding(&mut self, binding: &crate::Binding) -> BackendResult {
+        match *binding {
+            crate::Binding::BuiltIn(builtin) => {
+                write!(self.out, " : {}", builtin_str(builtin))?;
+            }
+            crate::Binding::Location { location, .. } => {
+                write!(self.out, " : {}{}", LOCATION_SEMANTIC, location)?;
             }
-            self.write_function(module, &ep.function, &func_ctx)?;
+        }
 
-            if index < module.entry_points.len() - 1 {
+        Ok(())
+    }
+
+    fn write_ep_input_struct(
+        &mut self,
+        module: &Module,
+        func: &Function,
+        stage: ShaderStage,
+    ) -> BackendResult {
+        if !func.arguments.is_empty() {
+            let struct_name = match stage {
+                ShaderStage::Vertex => self.namer.call_unique("VertexInput"),
+                ShaderStage::Fragment => self.namer.call_unique("FragmentInput"),
+                ShaderStage::Compute => self.namer.call_unique("ComputeInput"),
+            };
+
+            let mut members = Vec::with_capacity(func.arguments.len());
+
+            write!(self.out, "struct {}", &struct_name)?;
+            writeln!(self.out, " {{")?;
+
+            for arg in func.arguments.iter() {
+                let arg_name = if let Some(ref name) = arg.name {
+                    name
+                } else {
+                    "member"
+                };
+                let member = EpStructMember {
+                    name: self.namer.call_unique(arg_name),
+                    ty: arg.ty,
+                    binding: arg.binding.clone(),
+                };
+
+                write!(self.out, "{}", INDENT)?;
+                self.write_type(module, member.ty)?;
+                write!(self.out, " {}", &member.name)?;
+                if let Some(ref binding) = member.binding {
+                    self.write_binding(binding)?;
+                }
+                write!(self.out, ";")?;
                 writeln!(self.out)?;
+
+                members.push(member);
             }
+
+            writeln!(self.out, "}};")?;
+            writeln!(self.out)?;
+
+            self.ep_input = Some(EntryPointBinding {
+                stage,
+                name: struct_name,
+                members,
+            });
+        } else {
+            log::error!("Entry point without input arguments in stage {:?}", stage);
         }
 
         Ok(())
     }
 
+    /// Helper method used to write global variables
+    /// # Notes
+    /// Always adds a newline
+    fn write_global(&mut self, module: &Module, handle: Handle<GlobalVariable>) -> BackendResult {
+        let global = &module.global_variables[handle];
+        let inner = &module.types[global.ty].inner;
+
+        let register_ty = match *inner {
+            TypeInner::Image { .. } => "t",
+            TypeInner::Sampler { .. } => "s",
+            _ => "",
```

Comment:
should be expanded (possibly in a follow-up) to fill https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-variable-register